server/plugins/router/router.module.js

```diff
@@ -130,6 +131,33 @@ internals.registerRoutes = function(server, next) {
                 },
             },
         },
+        {
+            method: ['GET', 'POST'],
+            path: internals.paths.graphQL,
+            handler: {
+                proxy: {
+                    mapUri: function (req, cb) {
+                        return cb(
+                            null,
+                            `${internals.options.storeUrl}${req.path}`,
+                            Object.assign( // Add 'origin' and 'host' headers to request before proxying
```

Comment:
I originally had this as a nice arrow function with a spread operator, but this made tests fail due to a bug in [eslint/espree](https://github.com/eslint/espree/pull/426) which was only fixed a few months ago. Upgrading our version of espree by 4 major versions seemed prohibitively complicated to make this one change, so I think we should just stick with the weird syntax for now.

Reply:
Where is `espree` used?

Reply:
If it is just part of `eslint` you can try bumping it ahead to 6 if that resolves the issue with `espree`. `eslint` doesn't have any functional impact so upgrading it should be very low risk.

Reply:
From my investigation it would mean a serious upgrade of many of the `hapi` libraries, we use, which are pretty fundamental to CLI. I view it as high-risk.